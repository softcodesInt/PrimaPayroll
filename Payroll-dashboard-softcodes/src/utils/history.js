/* -------------------------------------------------------------------------- */
/*                             External Dependeny                             */
/* -------------------------------------------------------------------------- */
import { createBrowserHistory } from 'history';

export default createBrowserHistory();
