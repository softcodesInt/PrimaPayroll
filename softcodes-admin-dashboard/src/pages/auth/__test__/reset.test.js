/* -------------------------- External Dependencies ------------------------- */
import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { Formik } from 'formik';
import { HelmetProvider } from 'react-helmet-async';

/* -------------------------- Internal Dependencies ------------------------- */
import history from 'redux/history';
import Reset from '../reset';

const mockStore = configureStore([]);
describe('Reset Component', () => {
	afterEach(cleanup);
	let store;
	let component;
	beforeEach(() => {
		store = mockStore({
			user: {
				isLoading: false,
			},
		});
		store.dispatch = jest.fn();
		component = renderer.create(
			<HelmetProvider>
				<Provider store={store}>
					<Router history={history}>
						<Reset />
					</Router>
				</Provider>
			</HelmetProvider>
		);
	});
	it('Reset Renders Without Crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<HelmetProvider>
				<Provider store={store}>
					<Router history={history}>
						<Reset />
					</Router>
				</Provider>
			</HelmetProvider>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Reset matches data snapshot', () => {
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Reset should dispatch an action on button click', () => {
		renderer.act(() => {
			component.root.findByType(Formik).props.onSubmit({
				password: '123456',
			});
		});

		expect(store.dispatch).toHaveBeenCalledTimes(1);
	});
});
