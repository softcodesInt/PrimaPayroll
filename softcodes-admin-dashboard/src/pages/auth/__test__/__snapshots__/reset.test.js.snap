// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Component Reset matches data snapshot 1`] = `
<div
  className="sc-hKFyIo gSJtJX"
>
  <div
    className="grid_layout d-block d-md-grid"
  >
    <div
      className="grid-1 d-flex"
    >
      <div
        className="child__item container"
        data-testid="auth-layout"
      >
        <svg
          className="mb-4"
        >
          logos.png
        </svg>
        <div
          className="sc-eCApGN bxhqDO"
        >
          <h3>
            Create new password
          </h3>
          <p
            className="intro__text"
          >
            Set new password for your account
          </p>
          <form
            onSubmit={[Function]}
          >
            <div
              className="sc-bdnylx edIswL"
              data-testid="input"
            >
              <label
                htmlFor="Type New Password"
              >
                Type New Password
              </label>
              <input
                className="form-control"
                data-testid="inputs"
                id="password"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Type New Password"
                type="password"
                value=""
              />
              <button
                className="input-icon "
                onClick={[Function]}
                type="button"
              >
                <svg>
                  eye.svg
                </svg>
              </button>
            </div>
            <div
              className="sc-bdnylx edIswL"
              data-testid="input"
            >
              <label
                htmlFor="Confirm New Password"
              >
                Confirm New Password
              </label>
              <input
                className="form-control"
                data-testid="inputs"
                id="newPassword"
                name="newPassword"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Confirm New Password"
                type="password"
                value=""
              />
              <button
                className="input-icon "
                onClick={[Function]}
                type="button"
              >
                <svg>
                  eye.svg
                </svg>
              </button>
            </div>
            <button
              className="sc-dlnjPT eCNwlC button btn-developp btn-block"
              data-testid="button"
              disabled={false}
              type="submit"
              width=""
            >
               
              Create new password
               
            </button>
          </form>
        </div>
      </div>
    </div>
    <div
      className="grid-2 d-none d-md-block"
    >
      <img
        alt="layout banner"
        src="Body.png"
      />
    </div>
  </div>
</div>
`;
